node/lib/cmd/submodule.js

```diff
@@ -135,3 +150,126 @@ exports.executeableSubcommand = co.wrap(function *(args) {
         doSummary(false);
     }
 });
+
+const doFindCommand = co.wrap(function *(path, metaCommittish, subCommittish) {
+    const colors = require("colors");
+
+    const GitUtil          = require("../util/git_util");
+    const LogUtil          = require("../util/log_util");
+    const Open             = require("../util/open");
+    const SubmoduleFetcher = require("../util/submodule_fetcher");
+    const SubmoduleUtil    = require("../util/submodule_util");
+    const UserError        = require("../util/user_error");
+
+    // We need to resolve and validate the committishes and submodule path
+    // before calling `findMetaCommit` to do the real work.
+
+    const repo = yield GitUtil.getCurrentRepo();
+    const workdir = repo.workdir();
+
+    const subNames = yield SubmoduleUtil.getSubmoduleNames(repo);
+    const openSubNames = yield SubmoduleUtil.listOpenSubmodules(repo);
+
+    // Here, we find which submodule `path` refers too.  It migt be invalid by
+    // referring to no submodule, or by referring to more than one.
+
+    const relPath = yield GitUtil.resolveRelativePath(workdir,
+                                                      process.cwd(),
+                                                      path);
+    const resolvedPaths = yield SubmoduleUtil.resolvePaths(workdir,
+                                                           [relPath],
+                                                           subNames,
+                                                           openSubNames);
+    const paths = Object.keys(resolvedPaths);
```

Comment:
If the submodule is gone on HEAD, we won't be able to open it to resolve `subCommittish`; we'll fail immediately. We could, in theory, look at the submodules in `metaCommittish`, but while I'm OK with opening a submodule to make this work, I don't want to change the state of the repo otherwise.

(not sure if this answers your question)